### DISCUSSION TITLE
About Mongo Migrations

### DISCUSSION THREAD

**Discussion by Meerkap** [prisma/prisma#13425] [Q&A] [answered]
Hello, I just came with a question some days ago in StackOverflow but no one replied me, so i´m asking here:

This is the [post ](https://stackoverflow.com/questions/72257436/prisma-migrating-issue-with-mongo-schema-and-enums) i made.

I´ll clone the question here  in any case you won´t enter to the post.

> Post:

**So I´ve made this schema in first instance**

```js
    generator client {
      provider = "prisma-client-js"
    }
    
    datasource db {
      provider = "mongodb"
      url      = env("DATABASE_URL")
    }
    
    model User {
    
      id        String      @id @default(auto()) @map("_id") @db.ObjectId
      email     String      @unique
      role      TipoFichaje @default(JORNADA_ENTRADA)
      createdAt DateTime    @db.Date
    
    }
    
    model Entity {
    
      id        String   @id @default(auto()) @map("_id") @db.ObjectId
      name      String
      createdAt DateTime @db.Date
    
    }
    
    enum TipoFichaje {
      JORNADA_ENTRADA
      JORNADA_SALIDA
      DESAYUNO
      COMIDA
      FORMACION
      FORMACION3
    }
```

**Well then I push the db (prisma db push --preview-feature) and after that I Inserted an object like this:**

```js
    {
      email: "email@gmail.com"
      role:"FORMACION3"
      createdAt:1970-01-01T00:00:00.000+00:00
    }
```

**After that I remove the "Formacion3" ENUM from TipoFichaje and push again (prisma db push)
Like this:**

```js
    enum TipoFichaje {
          JORNADA_ENTRADA
          JORNADA_SALIDA
          DESAYUNO
          COMIDA
          FORMACION
    }
```

**When I try to get the data with Prisma Studio throws this error:**

```js
    Message: Error in Prisma Client request: 
    
    
    Invalid `prisma.user.findMany()` invocation:
    
    
      Value 'FORMACION3' not found in enum 'TipoFichaje'
      
    Query:
    {
      "modelName": "User",
      "operation": "findMany",
      "args": {
        "take": 100,
        "skip": 0,
        "select": {
          "id": true,
          "email": true,
          "role": true,
          "createdAt": true
        }
      }
    }
```

### My question is: Is it a normal behavior? I mean If i migrate the db with the new changes shouldn't migration fix this? How can I fix this?

**Comment by janpio** [comment, marked as answer]
MongoDB works different - you can have whatever you want in your database. Removing the `Formacion3` from the `enum` only makes sure you can not use it in Prisma any more, but Prisma itself does not go through the existing data and remove that value anywhere. For that you have to manually write a script that does that.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
